tests/pubsub_logging_test.py

```diff
@@ -41,6 +46,105 @@ def test_compat_urlsafe_b64encode(self):
         self.assertEqual(expected, result)
 
 
+class GetPubsubClientTest(unittest.TestCase):
```

Comment:
I think that unit tests are exactly what you describe and it's fine. Coverage for branches is important because we're using Python. It ensures that we don't have any TYPOs in any branches.
For example, a typo in except clause is overlooked if you don't actually cover the except clause.

We can later add end to end tests that checks if the code works as a whole.

WDYT?